sys-libs/libnbd/libnbd-1.14.2.ebuild

```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{8..11} )
+
+inherit autotools bash-completion-r1 python-single-r1
+
+DESCRIPTION="NBD client library in userspace"
+HOMEPAGE="https://gitlab.com/nbdkit/libnbd"
+SRC_URI="https://download.libguestfs.org/libnbd/$(ver_cut 1-2)-stable/${P}.tar.gz"
+
+LICENSE="LGPL-2.1+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="examples fuse gnutls go ocaml python test"
+
+RESTRICT="!test? ( test ) test? ( fuse? ( userpriv ) )"
+
+REQUIRED_USE="
+	python? ( ${PYTHON_REQUIRED_USE} )
+"
+RDEPEND="
+	examples? ( dev-libs/glib dev-libs/libev )
+	fuse? ( sys-fs/fuse:3 )
+	gnutls? ( net-libs/gnutls:= )
+	go? ( dev-lang/go )
+	ocaml? ( >=dev-lang/ocaml-4.03:=[ocamlopt] )
+	python? ( ${PYTHON_DEPS} )
+	dev-libs/libxml2
+"
+DEPEND="
+	${RDEPEND}
+	test? ( sys-block/nbd[gnutls?] )
+"
+BDEPEND="dev-lang/perl"
+
+DOCS=( README.md SECURITY TODO )
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+
+	eautoreconf -i
+}
+
+src_configure() {
+	local myeconfargs=(
+		$(use_enable fuse)
+		$(use_enable go golang)
+		$(use_enable ocaml)
+		$(use_enable python)
+		$(use_with gnutls)
+		--with-uri-support
+	)
+
+	export bashcompdir="$(get_bashcompdir)"
+
+	econf "${myeconfargs[@]}"
+}
+
+src_test() {
+	if use fuse ; then
+		[[ -e /dev/fuse ]] || die "/dev/fuse is required to run the test"
+		addwrite /dev/fuse
+	fi
```

Comment:
I started from other ebuilds that use it as dev-util/tup and sys-fs/unionfs-fuse and other others that use addwrite app-editors/vim , dev-db/sqlite , dev-lang/python , net-misc/openssh , etc..


I find it interesting that all possible tests are carried out, so that it can be verified that the system is correct for its operation, and it may also happen that more tests for that functionality are added later, although now there are only a few.

Reply:
Well, there's a few differences there.  Out of these, only tup and unionfs are based on fuse fundamentally.  Without fuse, neither of the two can be tested.  The other packages you listed add harmless devices, like ptmx.

Letting fragile code access /dev/fuse in Portage exposes a large risk, in terms of reliability, and the gain here is a few tests for a single plugin, so I don't think it's worth it.